slash/utils/cli_utils.py

```diff
@@ -128,24 +128,39 @@ def __init__(self, *args, **kwargs):
 COLOR_RESET = colorama.Fore.RESET + colorama.Back.RESET + colorama.Style.RESET_ALL  # pylint: disable=no-member
 
 
+class Styler(object):
+    should_colorize = None
+
+    def __init__(self, style):
+        self._style = style
+        self._str = None
+
+    def __call__(self, string):
+        if self.should_colorize:
+            return '{0}{1}{2}'.format(self._style, string, COLOR_RESET)
+        return string
+
 def make_styler(style):
-    return lambda s: '{0}{1}{2}'.format(style, s, COLOR_RESET)
+    return Styler(style)
 
 UNDERLINED = '\x1b[4m'
 
 
 class Printer(object):
 
-    def __init__(self, report_stread, enable_output=True):
+    def __init__(self, report_stread, enable_output=True, should_colorize=None):
         self._stream = report_stread
         self._output_enabled = enable_output
+        self._should_colorize = None
+        self.set_colors(should_colorize)
 
-    def _colored_print(self, *args):
-        print(*args, file=self._stream)
+    def set_colors(self, should_colorize):
+        self._should_colorize = should_colorize if should_colorize is not None else self._stream.isatty()
+        Styler.should_colorize = self._should_colorize
```

Comment:
See my above comments, and add to it a final method called `should_colorize` that takes everything into account:

```
should_color = self._force_color or (self._color_enabled and self._stream.isatty)
```